src/libraries/System.Private.Xml/src/System/Xml/Core/XsdValidatingReader.cs

```diff
@@ -796,11 +796,11 @@ public override bool ReadElementContentAsBoolean()
             {
                 if (xmlType != null)
                 {
-                    return xmlType.ValueConverter.ToBoolean(typedValue);
+                    return xmlType.ValueConverter.ToBoolean(typedValue!);
```

Comment:
I'll double check

Reply:
Seems to me that it may be null in some cases when XmlSchemaContentType is not TextOnly - it returns `_atomicValue` which I checked and it has direct `null` assignments in some states. I suspect there is a way for this to return null but it will likely end up in some sort of error.

Reply:
OK